### ISSUE TITLE
[Linux] Compiling/Running Android on AMD chip causes memory issue

### ISSUE THREAD

**Issue by bksubhuti** [flutter/flutter#136528]
### Steps to reproduce

I have had this problem many times.. Only now do I complain.
I have linux mint.  It does this in Ubuntu as well.

I have searched this issue but it is closed and locked.
https://github.com/flutter/flutter/issues/90844

When ever I try to compile apk's or debug (especially debug) I get a system locked up if I have Google chrome open.  This does not do this in Windows.  It was one of the reasons why I stopped using linux until I got a virus or something.  Now I'm back.. but I keep accidentally opening the browser.

The results are the system freezes and power down is needed.

```
System:
  Kernel: 5.15.0-86-generic x86_64 bits: 64 compiler: gcc v: 11.4.0 Desktop: Cinnamon 5.8.4
    tk: GTK 3.24.33 wm: muffin dm: LightDM Distro: Linux Mint 21.2 Victoria base: Ubuntu 22.04 jammy
Machine:
  Type: Laptop System: LENOVO product: 82KT v: IdeaPad 3 14ALC6 serial: <superuser required>
    Chassis: type: 10 v: IdeaPad 3 14ALC6 serial: <superuser required>
  Mobo: LENOVO model: LNVNB161216 v: SDK0T76465WIN serial: <superuser required> UEFI: LENOVO
    v: GLCN52WW date: 12/09/2022
Battery:
  ID-1: BAT0 charge: 14.8 Wh (35.3%) condition: 41.9/45.0 Wh (93.1%) volts: 11.0 min: 11.1
    model: SMP L20M3PF0 serial: <filter> status: Discharging
CPU:
  Info: 6-core model: AMD Ryzen 5 5500U with Radeon Graphics bits: 64 type: MT MCP arch: Zen 2
    rev: 1 cache: L1: 384 KiB L2: 3 MiB L3: 8 MiB
  Speed (MHz): avg: 1410 high: 1489 min/max: 1400/2100 boost: enabled cores: 1: 1397 2: 1397
    3: 1395 4: 1489 5: 1466 6: 1397 7: 1397 8: 1396 9: 1397 10: 1397 11: 1397 12: 1397
    bogomips: 50310
  Flags: avx avx2 ht lm nx pae sse sse2 sse3 sse4_1 sse4_2 sse4a ssse3 svm
Graphics:
  Device-1: AMD Lucienne vendor: Lenovo driver: amdgpu v: 6.2.4 pcie: speed: 8 GT/s lanes: 16
    ports: active: eDP-1 empty: HDMI-A-1 bus-ID: 03:00.0 chip-ID: 1002:164c
  Device-2: Chicony Integrated Camera type: USB driver: uvcvideo bus-ID: 1-3:2
    chip-ID: 04f2:b725
  Display: x11 server: X.Org v: 1.21.1.4 driver: X: loaded: amdgpu,ati
    unloaded: fbdev,modesetting,vesa gpu: amdgpu display-ID: :0 screens: 1
  Screen-1: 0 s-res: 1920x1080 s-dpi: 96
  Monitor-1: eDP res: 1920x1080 dpi: 158 diag: 354mm (13.9")
  OpenGL: renderer: AMD Radeon Graphics (renoir LLVM 16.0.6 DRM 3.54 5.15.0-86-generic)
    v: 4.6 Mesa 23.2.0-devel direct render: Yes
Audio:
  Device-1: AMD Renoir Radeon High Definition Audio vendor: Lenovo driver: snd_hda_intel v: kernel
    pcie: speed: 8 GT/s lanes: 16 bus-ID: 03:00.1 chip-ID: 1002:1637
  Device-2: AMD Raven/Raven2/FireFlight/Renoir Audio Processor vendor: Lenovo driver: N/A pcie:
    speed: 8 GT/s lanes: 16 bus-ID: 03:00.5 chip-ID: 1022:15e2
  Device-3: AMD Family 17h HD Audio vendor: Lenovo driver: snd_hda_intel v: kernel pcie:
    speed: 8 GT/s lanes: 16 bus-ID: 03:00.6 chip-ID: 1022:15e3
  Sound Server-1: ALSA v: k5.15.0-86-generic running: yes
  Sound Server-2: PulseAudio v: 15.99.1 running: yes
  Sound Server-3: PipeWire v: 0.3.48 running: yes
Network:
  Device-1: Realtek RTL8822CE 802.11ac PCIe Wireless Network Adapter vendor: Lenovo
    driver: rtw_8822ce v: N/A pcie: speed: 2.5 GT/s lanes: 1 port: 2000 bus-ID: 01:00.0
    chip-ID: 10ec:c822
  IF: wlp1s0 state: up mac: <filter>
Bluetooth:
  Device-1: Realtek Bluetooth Radio type: USB driver: btusb v: 0.8 bus-ID: 3-4:3
    chip-ID: 0bda:c123
  Report: hciconfig ID: hci0 rfk-id: 2 state: up address: <filter> bt-v: 3.0 lmp-v: 5.1
    sub-v: 6d7d
Drives:
  Local Storage: total: 238.47 GiB used: 94.25 GiB (39.5%)
  ID-1: /dev/nvme0n1 vendor: Samsung model: MZALQ256HBJD-00BL2 size: 238.47 GiB speed: 31.6 Gb/s
    lanes: 4 serial: <filter> temp: 30.9 C
Partition:
  ID-1: / size: 67.26 GiB used: 51.18 GiB (76.1%) fs: ext4 dev: /dev/nvme0n1p6
  ID-2: /boot/efi size: 256 MiB used: 39.5 MiB (15.4%) fs: vfat dev: /dev/nvme0n1p1
Swap:
  ID-1: swap-1 type: file size: 2 GiB used: 0 KiB (0.0%) priority: -2 file: /swapfile
Sensors:
  System Temperatures: cpu: N/A mobo: N/A gpu: amdgpu temp: 38.0 C
  Fan Speeds (RPM): N/A
Repos:
  Packages: apt: 2842
  No active apt repos in: /etc/apt/sources.list
  Active apt repos in: /etc/apt/sources.list.d/amdgpu-proprietary.list
    1: deb https: //repo.radeon.com/amdgpu/23.20/amdgpu/ubuntu jammy proprietary
  Active apt repos in: /etc/apt/sources.list.d/amdgpu.list
    1: deb https: //repo.radeon.com/amdgpu/23.20/amdgpu/ubuntu jammy main
  Active apt repos in: /etc/apt/sources.list.d/google-chrome.list
    1: deb [arch=amd64] https: //dl.google.com/linux/chrome/deb/ stable main
  Active apt repos in: /etc/apt/sources.list.d/official-package-repositories.list
    1: deb http: //packages.linuxmint.com victoria main upstream import backport
    2: deb http: //archive.ubuntu.com/ubuntu jammy main restricted universe multiverse
    3: deb http: //archive.ubuntu.com/ubuntu jammy-updates main restricted universe multiverse
    4: deb http: //archive.ubuntu.com/ubuntu jammy-backports main restricted universe multiverse
    5: deb http: //security.ubuntu.com/ubuntu/ jammy-security main restricted universe multiverse
  Active apt repos in: /etc/apt/sources.list.d/rocm.list
    1: deb [arch=amd64] https: //repo.radeon.com/amdgpu/23.20/rocm/apt/5.7 jammy main
  Active apt repos in: /etc/apt/sources.list.d/vscode.list
    1: deb [arch=amd64,arm64,armhf] http: //packages.microsoft.com/repos/code stable main
Info:
  Processes: 349 Uptime: 5m Memory: 5.63 GiB used: 1.71 GiB (30.4%) Init: systemd v: 249
  runlevel: 5 Compilers: gcc: 11.4.0 alt: 11/12 clang: 14.0.0-1ubuntu1.1 Client: Cinnamon v: 5.8.4
  inxi: 3.3.13

```

### Expected results

I expect to be able to run chrome browser and debug apk at the same time without the system freezing.

### Actual results

The system freezes and I have to power down and restart.

### Logs

<details><summary>Logs</summary>

```console
$ flutter --verbose create foo.
[  +42 ms] executing: uname -m
[   +5 ms] Exit code 0 from: uname -m
[        ] x86_64
[  +56 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required,
skipping update.
[   +2 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required,
skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping
update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping
update.
[        ] Artifact Instance of 'LegacyCanvasKitRemover' is not required,
skipping update.
[   +3 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required,
skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping
update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping
update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required,
skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required,
skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required,
skipping update.
[   +1 ms] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required,
skipping update.
[  +82 ms] Artifact Instance of 'MaterialFonts' is not required, skipping
update.
[        ] Artifact Instance of 'GradleWrapper' is not required, skipping
update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required,
skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required,
skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping
update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping
update.
[        ] Artifact Instance of 'LegacyCanvasKitRemover' is not required,
skipping update.
[        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[        ] Artifact Instance of 'WindowsEngineArtifacts' is not required,
skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping
update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping
update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required,
skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required,
skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required,
skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required,
skipping update.
[   +4 ms] Artifact Instance of 'IosUsbArtifacts' is not required, skipping
update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping
update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping
update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping
update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping
update.
[        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping
update.
[        ] Artifact Instance of 'PubDependencies' is not required, skipping
update.
[  +85 ms] "flutter create" took 199ms.
[   +3 ms] "foo." is not a valid Dart package name.
           
           See https://dart.dev/tools/pub/pubspec#name for more information.
[   +2 ms] 
           #0      throwToolExit
(package:flutter_tools/src/base/common.dart:10:3)
           #1      CreateBase.projectName
           (package:flutter_tools/src/commands/create_base.dart:332:9)
           #2      CreateCommand.runCommand
           (package:flutter_tools/src/commands/create.dart:299:62)
           <asynchronous suspension>
           #3      FlutterCommand.run.<anonymous closure>
           (package:flutter_tools/src/runner/flutter_command.dart:1297:27)
           <asynchronous suspension>
           #4      AppContext.run.<anonymous closure>
           (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #5      CommandRunner.runCommand
(package:args/command_runner.dart:212:13)
           <asynchronous suspension>
           #6      FlutterCommandRunner.runCommand.<anonymous closure>
           (package:flutter_tools/src/runner/flutter_command_runner.dart:339:9)
           <asynchronous suspension>
           #7      AppContext.run.<anonymous closure>
           (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #8      FlutterCommandRunner.runCommand
           (package:flutter_tools/src/runner/flutter_command_runner.dart:285:5)
           <asynchronous suspension>
           #9      run.<anonymous closure>.<anonymous closure>
           (package:flutter_tools/runner.dart:115:9)
           <asynchronous suspension>
           #10     AppContext.run.<anonymous closure>
           (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #11     main (package:flutter_tools/executable.dart:90:3)
           <asynchronous suspension>
           
           
[ +267 ms] ensureAnalyticsSent: 252ms
[        ] Running 0 shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 1

```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
flutter doctor -v
[✓] Flutter (Channel stable, 3.13.6, on Linux Mint 21.2 5.15.0-86-generic,
    locale en_US.UTF-8)
    • Flutter version 3.13.6 on channel stable at
      /home/bhante/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ead455963c (2 weeks ago), 2023-09-26 18:28:17 -0700
    • Engine revision a794cf2681
    • Dart version 3.1.3
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /home/bhante/Android/Sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: /home/bhante/development/android-studio/jbr/bin/java
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • Ubuntu clang version 14.0.0-1ubuntu1.1
    • cmake version 3.22.1
    • ninja version 1.10.1
    • pkg-config version 0.29.2

[✓] Android Studio (version 2022.3)
    • Android Studio at /home/bhante/development/android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b829.9-10027231)

[✓] VS Code (version 1.83.0)
    • VS Code at /usr/share/code
    • Flutter extension version 3.74.0

[✓] Connected device (3 available)
    • SM A505G (mobile) • R58M85Z9HXX • android-arm64  • Android 11 (API 30)
    • Linux (desktop)   • linux       • linux-x64      • Linux Mint 21.2
      5.15.0-86-generic
    • Chrome (web)      • chrome      • web-javascript • Google Chrome
      118.0.5993.70

[✓] Network resources
    • All expected network resources are available.

• No issues found!

```

</details>

**Comment by darshankawar** [comment]
Thanks for the report @bksubhuti 
Can you provide the details of the chrome browser version you are using ? Were you able to check on different chrome versions to confirm if the issue persists using those as well ?



> ```
>   +85 ms] "flutter create" took 199ms.
> [   +3 ms] "foo." is not a valid Dart package name.
>            
>            See https://dart.dev/tools/pub/pubspec#name for more information.
> ```

The error is expected as the name you are passing is not valid as mentioned in the log. You'll need to pass a valid name and check if it still persists or not.

**Comment by bksubhuti** [comment]
Just a quick jump on your playground.. i see that appbar does change to the proper pastel version of the primary color.. but I'm not getting it by default as expected.

**Comment by bksubhuti** [comment]
I should mention that this is only during compile time or "launching" time.  I just ran the emulator and had it load and then I opened the chrome browser and it did not crash.  Normally i use my phone becuase it is much slower on the emulator.  

This bug was more on the amd computer than my older intel.  I only have 8 gb of ram.   It does **not** do this on windows dual boot on the same laptop.

**Comment by huycozy** [comment]
Hey @bksubhuti 
I checked this issue on my machine with Ubuntu distro (`22.04.3 LTS 6.2.0-32-generic`) and using Intel chip but can't reproduce the error. Verifying the issue with Android device (Realme 6, Android 11).

The Android app can be built normally when the Google Chrome browser is open before. 

I saw it went well from your [bar.txt](https://github.com/flutter/flutter/files/12895221/bar.txt) output file above. Does that mean you no longer see the initial error? 

Could you try checking this on an Android native project instead? I guess there are some processes that consume system resources too much causing the system to freezes.

**Comment by bksubhuti** [comment]
Well.. like I said in my original post.. It is linked to my newer (but now old), computer with amd ryzen, and probably the graphics.  Can you test in an amd chipset?  AMD Ryzen 5 5500U with Radeon Graphics × 6  8gig ram

**Comment by huycozy** [comment]
@bksubhuti Unfortunately, I don't have a machine with AMD chipset. For testing with native app, just try a default/skeleton app (it's easy to create from Android Studio IDE) then let us know if you have more information. Thanks!

**Comment by bksubhuti** [comment]
It seems to be that I have 5.6 gb of ram showing as available .  and it does climb up with chrome running during a compile.  I have 8 gb and I guess some is stolen by the graphics which is embedded.   It could be good to make the compiling more efficient in memory.  It is really a burden to run the emulator even with most apps closed..  I'm using linux mint which is more efficient than ubuntu gnome (first time running).  

Take note that windows can handle this without crashing. 
I might upgrade my RAM.  At this point, if you change the developer minimum specifications for memory to 16gb then you can close the issue.  I'm not sure that is acceptable or not.

**Comment by huycozy** [comment]
@bksubhuti Does that mean Flutter app consumes too much memory which causes system to freeze? 

Could you try running Flutter app in profile mode and observe how much RAM is consumed by Flutter app? Please check this for Chrome browser as well.

**Comment by bksubhuti** [comment]
haha.. i just lost all my logging in this chrome window..  while I was building.

basically.. 
it was about 4.5 gb with up to 2.0 gb cache .. 2.1 swap.
empty was about 2.gb.  
I could not connect to the emulator.. then I ran a build for apk release and it went up to 5 gb and cache was like 1.3 gb (out of 6 availalbe).. the swap was up to 2.1 (full).

The crash is a memory problem for sure.
I have to do this later.. i have study.. 

It appears to me that the memory did not go down totally back to normal when I stopped everything.  
seems like the memory is still being held somewhere.  I have chrome open vsc and sysmon.. before it was less than 2gb.. after trying to debug, after a build of release.. mem is 4.6 cache 1.3 swap 1.5.. like my machine is more or less idle and not doing work.

**Comment by huycozy** [comment]
Thanks for the info but it's still not clear whether Chrome is one of the factors here. I guess it's just a case intended to consume enough memory to create this issue when building with Flutter later. This means you can try opening another application that consumes the same amount of memory. Can you test this? And can you try building Flutter app without opening other apps on the system?

You can also check this [issue](https://github.com/flutter/flutter/issues/73402) to see if it's similar as well.

**Comment by bksubhuti** [comment]
only text editor + sys monitor

+vsc        2.3 gb cache 3.7   635mb swap
+ emulator  2.9 gb cache 3.4  swap 1.1
+ debug     3.7 cache 2.7   swap  gb 2.1
debug     4.1 2.2 cache   2.1 swap  (another time)
highest is 4.2 gb..mem

**Comment by bksubhuti** [comment]
Perhaps this could be stress tested on vm?  
I suspect that if I compiled and ran this a couple of times, I would be running out of more memory.  
Try to look for leaks.  I felt like the memory usage was still there even after the debug session ended.

**Comment by huycozy** [comment]
>+vsc 2.3 gb cache 3.7 635mb swap

>emulator 2.9 gb cache 3.4 swap 1.1
debug 3.7 cache 2.7 swap gb 2.1
debug 4.1 2.2 cache 2.1 swap (another time)
highest is 4.2 gb..mem

Could you be more specific on this? Please try to provide clear information similar to the first comment at https://github.com/flutter/flutter/issues/73402#issue-780390527 which will help us determine whether this issue is due to the device or Flutter. (hint: please use `top` command, find Flutter app, and share the result)

Also, are you only seeing this issue when building for Android platform? Does it happen if building for Linux or Web target?

**Comment by bksubhuti** [comment]
This info .. was based on debugging apk with an android emulator I think comes from android studio.  but i use vsc.
I said, it does not happen with windows .. same apps running.

It also does not happen when running flutter debug linux desktop.  I mostly prefer to test debug in desktop because compile time and snappiness of the app .  Actually linux is best for desktop debugging over windows.. 3 x faster to compile and start.  and sqlite is 3x faster. (FOR DESKTOP)...   seeing realtime changes after saving a file are the same for both OS or not noticeable.

**Comment by huycozy** [comment]
Thank you for sharing. I understand that upgrading the RAM  may "solve" the problem temporarily. However, the purpose of the issue is to report a bug that belongs to Flutter. Thus, we need more data insights than just observations.

First of all, since you also suspect it might be due to Android Studio, you should clarify whether the issue will occur if you create an Android native skeleton app or not. Additionally, using the Android emulator also takes up a lot of memory, so leave it out of the test and use a physical device if you have one.

I'm not sure if you started an emulator via Android Studio or the command line. If you go through Android Studio, then you are running it at the same time and consuming a certain amount of memory. Please clarify this in the response.

If the issue appears on Android native app, I'm afraid that this is an Android Studio issue (possibly due to its memory management being bad on your Linux mint distro) and this is outside the scope of Flutter.

On the contrary, if the issue only occurs with the Flutter sample app, use `top` command to watch app memory usage and share it here (maybe a duplicate of https://github.com/flutter/flutter/issues/73402):

```
top -p pid
```

**Comment by bksubhuti** [comment]
I'm in the middle of finals week (8 days).  
I can give a report later.. if I forget please give a bump.
I forgot to do a skeleton app.  I'll try .  I do remember this also happening using my phone.  
Obviously, things work better on my phone, but I remember the computer crashing too with external device debugging.
It could also be my app which is sqlite db and has 600+mb datafile.. But sqlite should be able to manage it efficiently.  That was why we switched from flat files (and to flutter too).
http://github.com/bksubhuti/tipitaka-pali-reader
There is a db download on the readme page.

Please give me a few days.  I install AS but mostly for the licenses and virtual devices and flutter doctor complaints.
I use vsc.  It is much lighter.  All my activity is with vsc.

It does look like I can upgrade 16gb ram and 512gb sdd which is less than $100.  My computer was about $400 new and you get what you pay for.  But it still can do some heavy flutter db apps and video editing too.. It is just not so easy.  and I need to remember to not have chrome open.
I wonder how many flutter devs who release to all of the official app stores (5 OS's) actually have such low level hardware.
**But please note.. it is a Linux + AMD thing only.**
Linux on my older intel i5 8gb ram acer didn't crash (as I remember.. but it did struggle).  ($500 new during 2019).

**Comment by huycozy** [comment]
>It could also be my app which is sqlite db and has 600+mb datafile.. But sqlite should be able to manage it efficiently. That was why we switched from flat files (and to flutter too).

Please also make sure you check on Flutter skeleton app (counter app) or a [completed and minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example). The given source above is a complex production project so it's not easy for us to find the cause.

**Comment by bksubhuti** [comment]
It was actually worse.. You can see the app memory..  only vsc and system monitor are running.
 
![image](https://github.com/flutter/flutter/assets/18184280/86e74519-834f-4c6a-a2ec-8840c9ef3c59)

**Comment by huycozy** [comment]
Thanks, it's a bit more intuitive, but it would be more specific to know how much memory the Flutter app is taking up. Could you check it on `Processes` tab or with command line mentioned above?

**Comment by bksubhuti** [comment]
Looks like ~2gb
![image](https://github.com/flutter/flutter/assets/18184280/318a4d60-07ad-4cf4-afbd-9b799ef0f5cc)

**Comment by huycozy** [comment]
Labeling the issue for more insights. I summarize the issue below. 

### Issue summary

- The issue occurs when compiling apk's or debug (especially debug). It causes the system to freeze and power down is needed.
- It can be reproducible on a skeleton app
- Issue only appears on Linux Mint OS on a machine having `AMD Ryzen 5 5500U with Radeon Graphics × 6 8gig ram`. It doesn't happen on Windows dual boot on the same laptop. Meanwhile, Linux on an older `intel i5 8gb ram acer` didn't crash.
- System memory info: 
    - https://github.com/flutter/flutter/issues/136528#issuecomment-1776561584
    - https://github.com/flutter/flutter/issues/136528#issuecomment-1778320725

**Comment by gspencergoog** [comment]
Unfortunately, there isn't really enough information here for us to even determine if it has to do with Flutter tools, or just a combination of applications running which use up more RAM than the machine has.

We don't really see that there is anything we can do here to help without more definitive information that Flutter and its debugging tools are the cause.

We're therefore going to reluctantly close this issue for now. 
(Desktop Triage)

**Comment by bksubhuti** [comment]
Sorry to resurrect this issue.. but I figured it out.. There are java processes left in memory or left running.  So What I need to do , is remember to kill those processes before starting again.. simple memory panel plugin helped me figure this out.. So if I compile 2 or 3 times without killing the unkilled java processes.. then it is sure to run out of memory.. maybe even if I upgraded to 16 it would eat that up if I compiled a few times.  Would you like a screenshot.. This is from compiling apk only.  Using vsc or commandline in the vsc terminal.

**Comment by bksubhuti** [comment]
Here you can see the screenshot shows java still running after the app was built.
This will memory creep each time I do a compile command.. flutter build apk --release
I'm sure release or debug does not make a difference.
right now, I'm at 78% memory 
If I kill these processes.. I go back to 40+% or so.. which should make sense mathematically.


![image](https://github.com/flutter/flutter/assets/18184280/38784d5d-0e18-4ad0-ad49-e0356bb64e0d)

![image](https://github.com/flutter/flutter/assets/18184280/2a32d873-48f8-4619-bb2e-03cf35355068)


**after killing 2 java processes (no they don't increase in number of processes with each build ** 
It says 49% but I have chrome running
![image](https://github.com/flutter/flutter/assets/18184280/c8ab46d9-597d-4bb6-a4b3-1be9aa60ce0a)
